spec/features/users/recovery_code_spec.rb

```diff
@@ -69,49 +55,21 @@
       end
     end
 
-    context 'informational text' do
-      let(:accordion_selector) { generate_class_selector('accordion') }
-      let(:content_selector) { generate_class_selector('accordion-content') }
-
+    context 'recovery code actions and information' do
       before do
-        sign_in_and_2fa_user
+        @user = sign_in_and_2fa_user
         click_link t('profile.links.regenerate_recovery_code')
       end
 
-      scenario 'it displays the recovery code info header' do
-        expect(page).to have_content(t('users.recovery_code.help_text_header'))
-      end
-
-      context 'with javascript disabled' do
-        scenario 'content is visible by default' do
-          expect(page).to have_xpath("//#{accordion_selector}[@aria-expanded='true']")
-          expect(page).to have_xpath("//#{content_selector}[@aria-hidden='false']")
-          expect(page).to have_content(t('users.recovery_code.help_text'))
-        end
-      end
-
-      context 'with javascript enabled', js: true do
-        scenario 'content is hidden by default' do
-          expect(page).to have_xpath("//#{accordion_selector}[@aria-expanded='false']")
-          expect(page).not_to have_content(t('users.recovery_code.help_text'))
-
-          page.find('.accordion-header').click
-          expect(page).to have_xpath("//#{accordion_selector}[@aria-expanded='true']")
-          expect(page).to have_content(t('users.recovery_code.help_text'))
-        end
-      end
+      it_behaves_like 'recovery code page', @user
```

Comment:
I don't think the `@user` argument is necessary here because it's not being passed in to the shared examples. By setting `@user` in the before block, the shared examples have access to it.